app/Support/Charts.php

```diff
@@ -122,6 +121,50 @@ public static function rawAggregateByMonths(Builder $query, Carbon $start, strin
         return $dates->merge($results);
     }
 
+    public static function avgByHours(Builder $query, string $group, string $column = null, int $hours = 23)
+    {
+        return static::aggregateByHours($query, 'avg', $group, $column, $hours);
+    }
+
+    public static function aggregateByHours(Builder $query, string $function, string $group, string $column = null, int $hours = 23)
+    {
+        $carbon = Carbon::now();
+        $start = $carbon->subHours($hours);
+        $result = static::rawAggregateByHours($query, $start, $function, $group, $column);
+
+        return $result->mapWithKeys(function ($value, string $date) {
+            $carbon = Carbon::createFromFormat('Y-m-d H', $date);
+
+            return [$carbon->translatedFormat('Y-m-d H') => $value];
```

Comment:
Maybe find a way to improve the display format ? Because I find for example just `2021-04-03 8` to be confusing, we don't know whats the `8`